src/tools/rbd_mirror/image_sync/SnapshotCopyRequest.cc

```diff
@@ -354,8 +368,23 @@ void SnapshotCopyRequest<I>::send_snap_protect() {
     librados::snap_t remote_snap_id = *snap_id_it;
 
     m_remote_image_ctx->snap_lock.get_read();
+
```

Comment:
Minor: I don't think you'll need this for the unprotect/protect states since only user snapshots should be protected -- but I think you need to skip over non-user snapshots in the remove state machine for now.